lang/wasm.cc

```diff
@@ -70,9 +83,17 @@ static void init_abstract_arg(const abstract_type& t, const bytes_opt& param, st
         if (serialized_size > std::numeric_limits<int32_t>::max()) {
             throw wasm::exception(format("Serialized parameter is too large: {} > {}", param->size(), std::numeric_limits<int32_t>::max()));
         }
-        auto grown = memory.grow(store, 1 + (sizeof(int32_t) + serialized_size - 1) / WASM_PAGE_SIZE); // for fitting the serialized size + the buffer itself
-        if (!grown) {
-            throw wasm::exception(format("Failed to grow wasm memory to {}: {}", serialized_size, grown.err().message()));
+        switch (uint32_t abi_ver = get_abi(instance, store, data)) {
+            case 0: {
+                auto grown = memory.grow(store, 1 + (sizeof(int32_t) + serialized_size - 1) / WASM_PAGE_SIZE); // for fitting serialized size + the buffer itself
+                if (!grown) {
+                    throw wasm::exception(format("Failed to grow wasm memory to {}: {}", serialized_size, grown.err().message()));
+                }
+                mem_size = std::move(grown).unwrap() * WASM_PAGE_SIZE;
+                break;
+            }
```

Comment:
I think we can drop this old ABI since we have no users.

Reply:
I left the old ABI mostly for the examples in `test_wasm.py` - for each tested function there we have its wasm code in text format, which would get quite huge if each function included the implementation of malloc/free